dartsim/src/SimulationFeatures.hh

```diff
@@ -51,6 +51,12 @@ class SimulationFeatures :
 
   public: void Write(WorldPoses &_poses) const;
 
+  public: void Write(ChangedWorldPoses &_changedPoses) const;
+
+  public: void Write(Contacts &_contacts) const;
+
+  public: void Write(JointPositions &_jointPositions) const;
```

Comment:
Since the `SimulationFeatures` class now inherits `CanWriteExpectedData` instead of `CanWriteRequiredData` (`ChangedWorldPoses` is of type expected data), `Write` methods need to be implemented for all of the expected (and required) data types. All of these methods have an empty implementation except for `Write(ChangedWorldPoses &_changedPoses)`, but I don't think there's a better way to solve this problem of empty methods since we have to adhere to the interface.

Reply:
https://github.com/ignitionrobotics/ign-physics/pull/238/commits/1433ddbc73999fc149e25dc206448f4ee7b52ac2